0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,387 +110,380 @@ informative:
 
-## Condition {#condition-format}
-Below are the string and binary encoding formats for a condition.
+The advantage of using crypto-conditions for such use cases as opposed to a turing complete contract scripting language is the fact that the outcome of a crypto-condition validation is deterministic across platforms as long as the underlying cryptographic primitives are correctly implemented.
 
-### String Format {#string-condition-format}
+# Validation of a fulfillment
 
-Conditions are ASCII encoded as:
+Validation of a fulfillment (F) against a condition (C) and a message (M), in the majority of cases, the following steps. 
 
-    "cc:" BASE16(type) ":" BASE16(featureBitmask) ":"
-        BASE64URL(fingerprint) ":" BASE10(maxFulfillmentLength)
+1. The implementation must derive a condition from the fulfillment and ensure that the derived condition (D) matches the given condition (C). 
 
+2. If the fulfillment is a simple crypto-condition AND is based upon a signature scheme (such as RSA-PSS or ED25519) then any signatures in the fulfillment (F) must be verified, using the appropriate signature verification algorithm, against the corresponding public key, also provided in the fulfillment and the message (M) (which may be empty).
 
-### Binary Format {#binary-condition-format}
+3. If the fulfillment is a compound crypto-condition then the sub-fulfillments MUST each be validated. In the case of the PREFIX-SHA-256 type the sub-fulfillment MUST be valid for F to be valid and in the case of the THRESHOLD-SHA-256 type the number of valid sub-fulfillments must be equal or greater than the threshold defined in F.
 
-Conditions are binary encoded as:
+If the derived condition (D) matches the input condition (C) AND the boolean circuit defined by the fulfillment evaluates to TRUE then the fulfillment (F) fulfills the condition (C).
 
-    Condition ::= SEQUENCE {
-      type ConditionType,
-      featureBitmask INTEGER (0..MAX),
-      fingerprint OCTET STRING,
-      maxFulfillmentLength INTEGER (0..MAX)
-    }
+A more detailed validation algorithm for each crypto-condition type is provided with the details of the type later in this document. In each case the notation F.x or C.y implies; the decoded value of the field named x of the fulfillment and the decoded value of the field named y of the Condition respectively.
 
-    ConditionType ::= INTEGER {
-      preimageSha256(0),
-      rsaSha256(1),
-      prefixSha256(2),
-      thresholdSha256(3),
-      ed25519(4)
-    } (0..65535)
+## Subfulfillments
 
-### Fields {#condition-format-fields}
+In validating a fulfillment for a compound crypto-condition it is necessary to validate one or more sub-fulfillments per step 3 above. In this instance the condition for one or more of these sub-fulfillments is often not available for comparison with the derived condition. Implementations MUST skip the first fulfillment validation step as defined above and only perform steps 2 and 3 of the validation.
```

Comment:
This section is confusing without the having seen the crypto-condition type definitions.  I'd consider dropping and letting people read these details later on.